plugins/utils/which-key.nix

```diff
@@ -276,36 +521,28 @@ with lib;
             padding = mkSpacing cfg.window.padding;
             inherit (cfg.window) winblend;
           };
```

Comment:
I'm not sure, but I think these need to be gated behind checking `isDefined`; so far we've only checked that _some of them_ are defined.

If we accidentally read an undefined option value, the module system will throw "used but not defined".

I think this should be something like:

```nix
window =
let
  mkSpacing = ;# omitted

  setIfDef =
    name: fn: option:
    optionalAttrs option.isDefined { ${name} = fn option.value; };
in
  setIfDef "border" lib.id opt.window.border
  // setIfDef "position" lib.id opt.window.position
  // setIfDef "margin" mkSpacing opt.window.margin
  // setIfDef "padding" mkSpacing opt.window.padding
  // setIfDef "winblend" lib.id opt.window.winblend;
```

Reply:
Maybe add a (temporary) test to check what happens when `plugins.which-key.window` is partially defined.

Reply:
I had my personal config (that I sent the init.lua output of during troubleshooting in matrix) and the test for setting just the window.border and it worked fine.